### PR TITLE
New approval email template

### PR DESCRIPTION
**Pull request by epwinchell**
jira: https://projects.engineering.redhat.com/browse/SSP-1524

This PR introduces a new email template for Approval.

Old
Screen Shot 2020-05-20 at 4 29 52 PM

New
image

### CODE REVIEW

**Comment by epwinchell** [comment]
@hsong-rh Ready for initial testing/iterating

**Comment by hsong-rh** [comment]
@epwinchell Let me test it

File: src/main/java/com/redhat/management/approval/EmailBody.java [hunk 1]
```diff
@@ -162,15 +162,15 @@ private String getRequestContentLines(Map<String, Object> contents) {
             if (entry.getKey().equals(PARAMS_KEY))
                 continue;
   
-            String line = "<strong>" + customizeKey(entry.getKey()) + ":</strong>" + entry.getValue().toString() + "<br>";
+            String line = "<tr><th role="rowheader" scope="row">" + customizeKey(entry.getKey()) + ":</th></tr><td>" + entry.getValue().toString() + "</td></tr>";
```
Comment (hsong-rh) [inline src/main/java/com/redhat/management/approval/EmailBody.java:165]: Compile error: need to escape these double quote: `"<tr><th role=\"rowheader\" scope=\"row\">"`

File: src/main/java/com/redhat/management/approval/EmailBody.java [hunk 2]
```diff
@@ -162,15 +162,15 @@ private String getRequestContentLines(Map<String, Object> contents) {
...
             lines.append(line);
         }
         return lines.toString();
     }
 
     private String getParamsTable(HashMap<String, Object> params) {
         StringBuilder paramsTable = new StringBuilder(
-                "<table><tbody><tr><td><strong>Key</strong></td><td><strong>Value<strong></td></tr>\n");
+                "<table width: 100%; class="rh-data-table rh-m-bordered"><thead><tr><th>Key</th><th>Value</th></tr></thead><tbody>\n");
```
Comment (hsong-rh) [inline src/main/java/com/redhat/management/approval/EmailBody.java:174]: Same thing here: `class=\"rh-data-table rh-m-bordered\"`

**Comment by epwinchell** [comment]
@gmcculloug @sbuenafe-rh Please review

**Comment by sbuenafe-rh** [comment]
@epwinchell I would make the following changes:

1) Swap out the "Red Hat Insights" logo for "Red Hat Ansible Automation Platform" logo
2) Swap out the first sentence "The following Order requires your approval:" to "The following Automation Services Catalog order requires your approval:"
3) Reorder the summary labels (Ordered by, Order #, Order date, Order time, Product, Portfolio)
4) Hide the Parameters section in the template, when the approver clicks on the button "Process Request" the parameters information will be on that screen. 
5) Remove the "Request ID" below the Parameters section. It's already reference at the top of the email template. 
6) Swap out the footer "This email was sent by Red Hat Insights" for "This email was sent by Red Hat Ansible Automation Platform"

UX mock as reference: https://marvelapp.com/941d903/screen/67920368

File: src/main/java/com/redhat/management/approval/EmailBody.java [hunk 3]
```diff
@@ -162,15 +162,15 @@ private String getRequestContentLines(Map<String, Object> contents) {
             if (entry.getKey().equals(PARAMS_KEY))
                 continue;
   
-            String line = "<strong>" + customizeKey(entry.getKey()) + ":</strong>" + entry.getValue().toString() + "<br>";
+            String line = "<tr><th role=\"rowheader\" scope=\"row\">" + customizeKey(entry.getKey()) + ":</th><td>" + entry.getValue().toString() + "</td></tr>";
```
Comment (bzwei) [inline src/main/java/com/redhat/management/approval/EmailBody.java:165]: @hsong-rh Is it possible to move all the html tags to the template and see pure java code here? Same for the next change.

Comment (hsong-rh) [reply src/main/java/com/redhat/management/approval/EmailBody.java:165]: No, the request content are dynamic. Template only has static place holders.

Comment (hsong-rh) [reply src/main/java/com/redhat/management/approval/EmailBody.java:165]: Now the parameter table will be removed. If the request content's key are fixed, we can consider to move them into template and leave here to pure java codes.

**Comment by hsong-rh** [comment]
@epwinchell I checked the keys in request content are pretty static. We may move them into template file and populate them in codes.

File: src/main/java/com/redhat/management/approval/EmailBody.java (hunk 3 above)
Comment (hsong-rh) [reply src/main/java/com/redhat/management/approval/EmailBody.java:165]: @epwinchell Please remove https://github.com/RedHatInsights/approval-process/pull/51/files#diff-b621204f4af974b72aac40edf046513fL70 to fix the compilation error

**Comment by hsong-rh** [comment]
@epwinchell Let fix travis failure first and merge this PR. I will create another one to move content keys into template and reorder them as Susan's request.

**Comment by gmcculloug** [comment]
@sbuenafe-rh Per https://github.com/RedHatInsights/approval-process/pull/51#issuecomment-632178590 I think we are very close to merging this change and will get your other items in a follow up PR.

The one thing I do want to call out is that we should probably remove the "Manage email preferences" link at the bottom of the email since they have no effect on approval emails at the moment.  Seems misleading to me to include that.  Thoughts?

**Comment by sbuenafe-rh** [comment]
@gmcculloug we can removed the "Manage email preferences" link for now and later on we can bring it back when we have requirements for an approver to manage their email preferences of how they would like to be notified by Approval requests.

**Comment by epwinchell** [comment]
@bzwei @hsong-rh I think we're ready to merge. Thanks.

**Comment by hsong-rh** [comment]
@gmcculloug I uploaded the latest image

**Pull request merged**